bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -25,17 +32,65 @@
  *
  * @author Markus Rathgeb - Initial contribution and API
  * @author Mark Herwege - Added methods to find broadcast address(es)
+ * @author Stefan Triller - Converted to OSGi service with primary ipv4 conf
  */
-public class NetUtil {
+@Component(name = "org.eclipse.smarthome.network", property = { "service.config.description.uri=system:network",
+        "service.config.label=Network Settings", "service.config.category=system" })
+public class NetUtil implements NetworkAddressService {
 
+    private static final String PRIMARY_ADDRESS = "primaryAddress";
     private static final Logger LOGGER = LoggerFactory.getLogger(NetUtil.class);
 
-    private NetUtil() {
+    private static final Pattern IPV4_PATTERN = Pattern
+            .compile("^(([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.){3}([01]?\\d\\d?|2[0-4]\\d|25[0-5])$");
+
+    private String primaryAddress;
+
+    @SuppressWarnings("unchecked")
+    protected void activate(ComponentContext componentContext) {
```

Comment:
Instead of pulling the properties out of the context, why don't you simply implement `activate(Map<String, Object> props)`?